core/src/main/java/org/testcontainers/RequireContainerSupportRule.java

```diff
@@ -0,0 +1,38 @@
+package org.testcontainers;
+
+import org.junit.AssumptionViolatedException;
+import org.junit.Rule;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+/**
+ * Junit 4 {@code Rule} that will skip the test if the docker client is not available.
+ * This rule must have a lower {@link Rule#order} specified that any other {@code Rule} that
+ * uses docker in order to skip tests, otherwise the other rules would fail first causing build failures.
+ * e.g. <pre><code>
+    {@literal @}Rule(order = -10)
+    public RequireContainerSupportRule rcr = new RequireContainerSupportRule();
+ }
+ * </code></pre>
+ */
+public class RequireContainerSupportRule implements TestRule {

```

Comment:
I went round the houses on the name of this a few times - in reality not sure if you require `docker` (the cli) at all, or just a configuration for docker (which could come from podman etc.)

Reply:
We can use docker, similar to what we have for Testcontainers annotation for JUnit Jupiter. See https://github.com/testcontainers/testcontainers-java/blob/de1324ed2800eff4da326d0c23d281399d006bc0/modules/junit-jupiter/src/main/java/org/testcontainers/junit/jupiter/Testcontainers.java#L65

Reply:
I wonder if `DockerCheckRule` make sense. I know naming is hard :)